### ISSUE TITLE
Notify extra installation steps`

### ISSUE THREAD

**Issue by Jarlakxen** [laughedelic/atom-ide-scala#35]
I think it could be very usefull, from a user experience perspective, if the plugin detect if `ScalametaLanguageServerPlugin` is not present or if the user didn't run `semanticdbEnable`. I could be just a notification with a link to [Metals global setup](https://github.com/scalameta/metals/blob/master/BETA.md).
Lots of the Atom plugin, if they require extra instalation steps, at least you recieve a notification about that.

**Comment by laughedelic** [comment]
This is a good idea! The question is how to detect that the user hasn't setup the project well (besides the server failure which could happen for different reasons).

**Comment by Jarlakxen** [comment]
I think checking for `~/.sbt/0.13/plugins/ScalametaLanguageServerPlugin.scala` or `~/.sbt/1.0/plugins/ScalametaLanguageServerPlugin.scala` could be an option for now. 
 One thing I noticed is that if you forgot to run `sbt semanticdbEnable`, the server start and it looks like is throwing an error and `atom-languageclient` is wrapping the error in a warning with a strange message. Probably the `atom-languageclient` is not parsing the error correctly or the error is in a wrong format. So, the Atom user isn't getting a notification, the server doesn't exit and nothing works.
This issue could be a solution for detect if the user forgot to run  `sbt semanticdbEnable` in the project.

**Comment by laughedelic** [comment]
> I think checking for ~/.sbt/0.13/plugins/ScalametaLanguageServerPlugin.scala or ~/.sbt/1.0/plugins/ScalametaLanguageServerPlugin.scala could be an option for now.

I'm not sure it's a good solution. There's a plan to provide an sbt plugin for metals setup: https://github.com/scalameta/metals/issues/178. At the moment copy-pasted `ScalametaLanguageServerPlugin` is a temporary solution and I feel that it would be wrong to encourage users to use it as something obligatory. Probably I'm wrong, I need to think a bit more about it.

How about checking project's `target/` for the presence of any `.semanticdb` files?

> One thing I noticed is that if you forgot to run sbt semanticdbEnable, the server start and it looks like is throwing an error and atom-languageclient is wrapping the error in a warning with a strange message. Probably the atom-languageclient is not parsing the error correctly or the error is in a wrong format.

I never used `semanticdbEnable` and I haven't seen such error/warning. Could you make a screenshot of how it looks?

**Comment by olafurpg** [comment]
One way to detect if semanticdbEnable/scalametaEnableCompletions was run is the presence of .compilerconfig files. If none exist then the plugin could senda messageRequest with link to installation docs, and button to "Don't remind me again".

**Comment by laughedelic** [comment]
@olafurpg thanks for the input. I think we should do it at the server side: do the checks and send a `showMessage` notification with a warning that some features won't work if the user won't setup the project well (and a link to the docs). I think it would be equally useful feedback for any client.

**Comment by olafurpg** [comment]
I agree, this should be handled by the server since that's where we look for .compilerconfig files.

**Comment by Jarlakxen** [comment]
Should I move this issue  to metals? =)

**Comment by laughedelic** [comment]
@Jarlakxen yes, please 👍
